### ISSUE TITLE
Tests to write/cleanup

### ISSUE THREAD

**Issue by domenic** [jackbsteinberg/get-originals-rewriter#15]
After #14 lands:

- There are identical files named replace-constructors.*.js and replace-constructors-js.*.js
- We should add a test showing that the originals constructors replacement still works if you alias
- We should add a test showing that the originals constructors replacement still works if you pass it to a function and use `new X()` inside the function, as long as `X` has an appropriate jsdoc markup.

**Comment by jackbsteinberg** [comment]
That test will come in the next PR

**Comment by jackbsteinberg** [comment]
Update: `ts-morph` is having a lot of trouble figuring out the case where a constructor is passed to a function, so the third bullet point will be left as polish if we decide it becomes necessary.

**Comment by domenic** [comment]
> Update: ts-morph is having a lot of trouble figuring out the case where a constructor is passed to a function, so the third bullet point will be left as polish if we decide it becomes necessary.

I did a bit more digging. The issue is basically there's no reasonable type for the constructors. E.g. in the snippet

```js
const y = Document;
```

what is the type of `y`? https://ts-ast-viewer.com/ indicates that it is (something that stringifies as) `{ new (): Document; prototype: Document; }`.

And indeed if I put as the input file

```js
/**
 * @param {{ new (): Document }} api - the api to call
 */
function f(api) {
  const a = new api();
  return a;
}

const output = f(Document);
```

then TypeScript will figure it out. So I guess we could add such a test...

**Comment by jackbsteinberg** [comment]
Is that something we want to test for? Or does it make more sense to decide not to pass Global constructors into methods which:

1. Doesn't make much sense anyway, if they're global then by definition you can access them from anywhere.

2. Is probably a style we already follow, just has yet to be formalized.

**Comment by domenic** [comment]
Yeah, I'd be fine declaring it out of scope. It's just that I think it'll work already anyway with the technique we're using, so adding a test might be useful to ensure we don't break it. I'm OK either way.
